src/python/grpcio/grpc/experimental/aio/_call.py

```diff
@@ -199,64 +186,32 @@ def time_remaining(self) -> Optional[float]:
-        return (await self._status).trailing_metadata()
+        return (await self._cython_call.status()).trailing_metadata()
 
     async def code(self) -> grpc.StatusCode:
-        await self._status
-        return self._code
+        cygrpc_code = (await self._cython_call.status()).code()
+        return _common.CYGRPC_STATUS_CODE_TO_STATUS_CODE[cygrpc_code]
 
     async def details(self) -> str:
-        return (await self._status).details()
+        return (await self._cython_call.status()).details()
 
     async def debug_error_string(self) -> str:
-        return (await self._status).debug_error_string()
-
-    def _set_initial_metadata(self, metadata: MetadataType) -> None:
-        self._initial_metadata.set_result(metadata)
-
-    def _set_status(self, status: cygrpc.AioRpcStatus) -> None:
-        """Private method to set final status of the RPC.
-
-        This method may be called multiple time due to data race between local
-        cancellation (by application) and Core receiving status from peer. We
-        make no promise here which one will win.
-        """
-        # In case of local cancellation, flip the flag.
-        if status.details() is _LOCAL_CANCELLATION_DETAILS:
-            self._locally_cancelled = True
-
-        # In case of the RPC finished without receiving metadata.
-        if not self._initial_metadata.done():
-            self._initial_metadata.set_result(_EMPTY_METADATA)
-
-        # Sets final status
-        self._status.set_result(status)
-        self._code = _common.CYGRPC_STATUS_CODE_TO_STATUS_CODE[status.code()]
+        return (await self._cython_call.status()).debug_error_string()
 
     async def _raise_for_status(self) -> None:
-        if self._locally_cancelled:
+        if self._cython_call.is_locally_cancelled():
             raise asyncio.CancelledError()
-        await self._status
-        if self._code != grpc.StatusCode.OK:
-            raise _create_rpc_error(await self.initial_metadata(),
-                                    self._status.result())
+        code = await self.code()
+        if code != grpc.StatusCode.OK:
+            raise _create_rpc_error(await self.initial_metadata(), await
+                                    self._cython_call.status())
 
     def _repr(self) -> str:
         """Assembles the RPC representation string."""
-        if not self._status.done():
-            return '<{} object>'.format(self.__class__.__name__)
-        if self._code is grpc.StatusCode.OK:
-            return _OK_CALL_REPRESENTATION.format(
-                self.__class__.__name__, self._code,
-                self._status.result().details())
-        else:
-            return _NON_OK_CALL_REPRESENTATION.format(
-                self.__class__.__name__, self._code,
-                self._status.result().details(),
-                self._status.result().debug_error_string())
+        return '<{} object>'.format(self.__class__.__name__)
```

Comment:
I didn't remember to add a comment to the PR telling that by moving the status logic under the Cython code made the `__repr__`  more difficult to reproduce since we do not have a synchronous method for getting the `code`. The Python layer uses always the coro `AioCall.status` wich does not suit well with this use case.

Before start trying to fix this, I wanted to ask if this was a hard requirement since it would imply a none negligible effort.

Reply:
This is a small plus for usability. Alternatively, we might move the entire `__repr__` logic down to Cython layer. Will that be more feasible?

Reply:
I will take care of this, but most likely I would be forced to move the __repr__ to the Cython layer.